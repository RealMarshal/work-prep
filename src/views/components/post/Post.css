.post {
  display: inline-block;
  max-width: 300px;
  margin: 1rem;
}