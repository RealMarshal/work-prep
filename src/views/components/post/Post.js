import React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'

import './Post.css'

const Post = ({title, text}) => (
  <Card className='post'>
    <CardContent className='post__content'>
      <Typography variant='title' className='post__title'>{ title }</Typography>
      <Typography className='post__text'>{ text }</Typography>
    </CardContent>
  </Card>
)

export default Post