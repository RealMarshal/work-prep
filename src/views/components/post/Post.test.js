import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Post from './Post'

describe('Post', () => {

  const testPost = {
    title: 'qewrlfda',
    text: 'asdmdamsdower'
  }
  const rendered = shallow(<Post title={testPost.title} text={testPost.text} />)

  it('should render correctly', () => {
    expect(toJson(rendered)).toMatchSnapshot()
  })

  it('should have title', () => {
    expect(rendered.find('.post__title').render().text()).toEqual(testPost.title)
  })

  it('should have text', () => {
    expect(rendered.find('.post__text').render().text()).toEqual(testPost.text)
  })
})