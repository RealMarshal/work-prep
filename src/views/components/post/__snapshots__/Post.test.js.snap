// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render correctly 1`] = `
<WithStyles(Card)
  className="post"
>
  <WithStyles(CardContent)
    className="post__content"
  >
    <WithStyles(Typography)
      className="post__title"
      variant="title"
    >
      qewrlfda
    </WithStyles(Typography)>
    <WithStyles(Typography)
      className="post__text"
    >
      asdmdamsdower
    </WithStyles(Typography)>
  </WithStyles(CardContent)>
</WithStyles(Card)>
`;
